gnd/src/androidTest/java/com/google/android/gnd/AddFeatureTest.java

```diff
@@ -0,0 +1,152 @@
+import com.google.android.gnd.persistence.remote.FakeRemoteStorageManager;
+import com.google.android.gnd.persistence.remote.RemoteDataStore;
+import com.google.android.gnd.persistence.remote.RemoteStorageManager;
+import com.google.android.gnd.persistence.remote.RemoteStorageModule;
+import com.google.android.gnd.persistence.uuid.FakeUuidGenerator;
+import com.google.android.gnd.persistence.uuid.OfflineUuidGenerator;
+import com.google.android.gnd.system.auth.AuthenticationManager;
+import com.google.android.gnd.system.auth.AuthenticationModule;
+import com.google.android.gnd.system.auth.FakeAuthenticationManager;
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import dagger.hilt.InstallIn;
+import dagger.hilt.android.components.ApplicationComponent;
+import dagger.hilt.android.qualifiers.ApplicationContext;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.UninstallModules;
+import javax.inject.Singleton;
+import org.junit.Rule;
+import org.junit.Test;
+
+@UninstallModules({AuthenticationModule.class, RemoteStorageModule.class,
+  LocalDatabaseModule.class})
+@HiltAndroidTest
+public class AddFeatureTest {
+
+  // Ensures that the Hilt component is initialized before running the ActivityScenarioRule.
+  public @Rule(order = 0)
+  HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+  // Sets the preferences so no login is required and an active project is selected.
+  public @Rule(order = 1)
+  SetPreferencesRule preferencesRule = new SetPreferencesRule();
+
+  public @Rule(order = 2)
+  ActivityScenarioRule scenarioRule =
+    new ActivityScenarioRule(MainActivity.class);
+
+  @Module
+  @InstallIn(ApplicationComponent.class)
+  abstract class TestDependenciesModule {
+
+    @Binds
+    @Singleton
+    abstract AuthenticationManager bindAuthenticationManager(
+      FakeAuthenticationManager authenticationManager);
+
+    @Binds
+    @Singleton
+    abstract RemoteDataStore bindRemoteDataStore(FakeRemoteDataStore remoteDataStore);
+
+    @Binds
+    @Singleton
+    abstract RemoteStorageManager bindRemoteStorageManager(
+      FakeRemoteStorageManager remoteStorageManager);
+
+    @Binds
+    @Singleton
+    abstract OfflineUuidGenerator offlineUuidGenerator(FakeUuidGenerator uuidGenerator);
+  }
+
+  @InstallIn(ApplicationComponent.class)
+  @Module
+  public class LocalDatabaseModule {
+
+    @Provides
+    @Singleton
+    LocalDatabase localDatabase(@ApplicationContext Context context) {
+      return Room.inMemoryDatabaseBuilder(context, LocalDatabase.class).build();
+    }
+  }
+
+  // Given: a logged in user - with an active project with no map markers.
+  // When: they tap on the centre of the map.
+  // Then: nothing happens - the feature fragment is not displayed.
+  @Test
+  public void tappingCrosshairOnEmptyMapDoesNothing() throws InterruptedException {
```

Comment:
Weird throws here, is it needed?

Reply:
I used to have a `Thread.sleep` in that method but it was removed so the `InterruptedException` can also be removed. Good spot, thanks.